### PR TITLE
feat : added Homepage functionality

### PR DESCRIPTION
**Pull request by Priya-Patel31**
Added few functionalities which have been listed below :

- like post 
- bookmark post 
- edit post 
- delete post 
- follow user

### CODE REVIEW

**Review by tushartiwari7 — APPROVED** [review]
Liked that you are using typescript.
code looks well structured with good namings.
Easily Understandable.
Good to merge.

File: src/features/posts/PostsSlice.ts [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+} from "firebase/firestore";
+import { db } from "../../firebaseApp";
+import { query, where } from "firebase/firestore";
+import {
+  BookmarkParams,
+  LikePostParams,
+  Post,
+  PostsInitialState,
+} from "./posts.types";
+
```
Comment (Ankur9669) [inline src/features/posts/PostsSlice.ts:21]: fix: Use Barrel exports

File: src/features/users/usersSlice.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import { createAsyncThunk, createSlice, PayloadAction } from "@reduxjs/toolkit";
+import { collection, getDocs, query, where } from "firebase/firestore";
+import { db } from "../../firebaseApp";
+import { User } from "../auth/auth.types";
+import { followUser, followUserReturnType } from "../auth/authSlice";
+import { usersInitialState } from "./users.types";
+
```
Comment (Ankur9669) [inline src/features/users/usersSlice.ts:7]: fix: Use Barrel exports

File: src/pages/authentication/signup/Signup.tsx [hunk 3]
```diff
@@ -15,7 +15,6 @@ import { AppDispatch } from "../../../app/store";
 import { toast } from "react-toastify";
 import { unwrapResult } from "@reduxjs/toolkit";
```
Comment (Ankur9669) [inline src/pages/authentication/signup/Signup.tsx:17]: Same

File: src/pages/authentication/signup/Signup.tsx [hunk 4]
```diff
@@ -15,7 +15,6 @@ import { AppDispatch } from "../../../app/store";
...
```
Comment (Ankur9669) [inline src/pages/authentication/signup/Signup.tsx:16]: What is this used for??

File: src/pages/authentication/signup/Signup.tsx [hunk 5]
```diff
@@ -15,7 +15,6 @@ import { AppDispatch } from "../../../app/store";
...
 
-
 export const Signup = () => {
   const navigate = useNavigate();
   const dispatch = useDispatch<AppDispatch>();
```
Comment (Ankur9669) [inline src/pages/authentication/signup/Signup.tsx:20]: You can use the hook useAppDispatch which is defined in app/hooks file

File: src/shared/createPost/CreatePost.tsx [hunk 6]
```diff
@@ -1,21 +1,96 @@
...
+  SetStateAction,
+} from "react";
 import "./createPost.css";
 import Picker, { SKIN_TONE_LIGHT } from "emoji-picker-react";
 import Avatar from "../../assets/images/avatar.jpg";
+import { editPost, uploadPost } from "../../features/posts/PostsSlice";
+import { Post } from "../../features/posts/posts.types";
+import { useAppDispatch, useAppSelector } from "../../app/hooks";
+import { unwrapResult } from "@reduxjs/toolkit";
+import { toast } from "react-toastify";
```
Comment (Ankur9669) [inline src/shared/createPost/CreatePost.tsx:16]: Same use barrel exports

File: src/shared/createPost/CreatePost.tsx [hunk 7]
```diff
@@ -1,21 +1,96 @@
...
+  };
+  const { user } = useAppSelector((state) => {
+    return state.auth;
+  });
+  const { uploadPostStatus } = useAppSelector((state) => {
+    return state.posts;
+  });
 
-    setInput(input + emojiObject.emoji)
+  const dispatch = useAppDispatch();
```
Comment (Ankur9669) [inline src/shared/createPost/CreatePost.tsx:45]: Nice that you have used the hook here

**Review by Ankur9669 — APPROVED** [review]
Very nicely done, the code looks awesome and the animation on create post modal is also looking awesome

File: src/pages/authentication/signup/Signup.tsx (hunk 4 above)
Comment (Priya-Patel31) [reply src/pages/authentication/signup/Signup.tsx:16]: for unwrapping the result and basically it is used for error handling and also checking the result after the dispatch .

File: src/features/users/usersSlice.ts (hunk 2 above)
Comment (Priya-Patel31) [reply src/features/users/usersSlice.ts:7]: Refactoring part I will do later.

File: src/pages/authentication/signup/Signup.tsx (hunk 5 above)
Comment (Priya-Patel31) [reply src/pages/authentication/signup/Signup.tsx:20]: done

**Pull request merged**